src/Server/RestRouterHandlers/SearchHandler.cpp

```diff
@@ -0,0 +1,219 @@
+        response.setStatusAndReason(HTTPResponse::HTTPStatus(http_status));
+        *response.send() << error_msg << std::endl;
+        return error_msg;
+    };
+
+    String data = "{}";
+    auto size = getContentLength();
+    if (size > 0)
+    {
+        data.resize(size);
+        input.readStrict(data.data(), size);
+    }
+    Poco::JSON::Parser parser;
+    auto payload = parser.parse(data).extract<Poco::JSON::Object::Ptr>();
+
+    String error;
+    if (!validatePayload(payload, error))
+        return send_error(jsonErrorResponse(error, ErrorCodes::BAD_REQUEST_PARAMETER));
+
+    const auto & query = getQuery(payload);
+    if (!validateQuery(query, error))
+        return send_error(jsonErrorResponse(error, ErrorCodes::BAD_REQUEST_PARAMETER));
+
+    LOG_INFO(log, "Execute query: {}", query);
+    ReadBufferFromString in{query};
+
+    /// Prepare output buffer
+    const auto out = getOutputBuffer(response);
+    try
+    {
+        /// FIXME: implement outbound solution to support query progress
+        executeQuery(in, *out, /* allow_into_outfile = */ false, query_context, {});
+    }
+    catch (Exception & e)
+    {
+        tryLogCurrentException(log);
+        /// Send the error message into already used (and possibly compressed) stream.
+        /// Note that the error message will possibly be sent after some data.
+        /// Also HTTP code 200 could have already been sent.
+        http_status = HTTPResponse::HTTP_INTERNAL_SERVER_ERROR;
+        response.setStatusAndReason(HTTPResponse::HTTPStatus(http_status));
+        bool data_sent = out->count() != out->offset();
+
+        /// If buffer has data, and that data wasn't sent yet, then no need to send that data
+        if (!data_sent)
+            out->position() = out->buffer().begin();
+
+        writeString(jsonErrorResponse(e.message(), e.code()), *out);
+        writeChar('\n', *out);
+        out->next();
+        out->finalize();
+    }
+
+    return "";
+}
+
+String SearchHandler::getQuery(const Poco::JSON::Object::Ptr & payload) const
+{
+    /// Setup query settings
+    const auto & mode = payload->has("mode") ? payload->get("mode").toString() : "standard";
+
+    /// FIXME: to support 'realtime' mode in future
+    /// FIXME: change the behavior of 'verbose' mode. Currently in frontier service, 'standard' and 'verbose' mode are the same
+    if (mode == "standard" || mode.empty() || mode == "verbose")
+        query_context->setDefaultFormat("JSONCompactEachRowWithNamesAndTypes");
+
+    const auto & start_time = payload->has("start_time") ? payload->get("start_time").toString() : "";
+    if (!start_time.empty())
+        query_context->setTimeParamStart(start_time);
+
+    const auto & end_time = payload->has("end_time") ? payload->get("end_time").toString() : "";
+    if (!end_time.empty())
+        query_context->setTimeParamEnd(end_time);
+
+    query_context->setSetting("unnest_subqueries", true);
```

Comment:
I afraid that we cannot remove this flag. Because to support paging, it still wrap the original query with "SELECT * from (original_query) ...", therefore it still need to optimize subquery later. Otherwise I have to rewrite the query in SearchHandler which is quite ineffient.

Reply:
I didn't say `remove`. I said instead of hard code it here, we will need make it configurable via a HTTP rest URI parameter. Otherwise if there are some bugs, we can't turn off this flag and need rebuild Daisy. Let's do this in the next PR since we are going to support more URI parameters anyway in the next PR.

Reply:
Got it.